expression/schema.go

```diff
@@ -102,9 +129,33 @@ func (s *Schema) Append(col *Column) {
 	s.Columns = append(s.Columns, col)
 }
 
+// SetUniqueKeys will set the value of Schema.Keys.
+func (s *Schema) SetUniqueKeys(keys []KeyInfo) {
+	s.Keys = keys
+}
+
+// GetColumnsIndices will return a slice which contains the position of each column in schema,
+// and a boolean which tell you whether all columns are found.
+func (s Schema) GetColumnsIndices(cols []*Column) (ret []int) {
```

Comment:
Because this function is called by a tempory schema, it will return the offsets.